### PR TITLE
[Merged by Bors] - feat: if one set is clopen in another, then it is the union of connected components in the larger set

### PR DESCRIPTION
**Pull request by j-loreaux**
---

- [x] depends on: #15144

I'm happy to put `isClopen_preimage_val` somewhere else, but I'm simply not sure where it should go.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Connected/Basic.lean [hunk 1]
```diff
@@ -3,7 +3,7 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl, Mario Carneiro, Yury Kudryashov
 -/
-import Mathlib.Data.Set.Image
+import Mathlib.Data.Set.Subset
```
Comment (semorrison) [inline Mathlib/Topology/Connected/Basic.lean:6]: Adding an import that is only used after line 1000! :-) could this file be split?

Comment (j-loreaux) [reply Mathlib/Topology/Connected/Basic.lean:6]: In fairness, it's a single file import (including transitive)

Comment (j-loreaux) [reply Mathlib/Topology/Connected/Basic.lean]: #15144

File: Mathlib/Topology/Connected/Clopen.lean [hunk 2]
```diff
@@ -121,6 +122,16 @@ theorem IsClopen.eq_univ [PreconnectedSpace α] {s : Set α} (h' : IsClopen s) (
     s = univ :=
   (isClopen_iff.mp h').resolve_left h.ne_empty
 
+open Set.Notation in
+lemma isClopen_preimage_val {X : Type*} [TopologicalSpace X] {u v : Set X}
+    (hu : IsOpen u) (huv : frontier u ∩ v = ∅) : IsClopen (v ↓∩ u) := by
```
Comment (Ruben-VandeVelde) [inline Mathlib/Topology/Connected/Clopen.lean:127]: Is `frontier u ∩ v = ∅` the best way to state this? Do we not have a disjointness predicate?

Comment (j-loreaux) [reply Mathlib/Topology/Connected/Clopen.lean:127]: Thanks, good point, I'll make that change.

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**